src/Orleans/Streams/Internal/StreamSubscriptionHandleImpl.cs

```diff
@@ -121,13 +115,6 @@ public override Task<StreamSubscriptionHandle<T>> ResumeAsync(IAsyncObserver<T>
 
             await NextItem(item, currentToken);
 
-            // check again, in case the expectedToken was changed indiretly via ResumeAsync()
-            if (expectedToken != null)
```

Comment:
But I thought you wanted that code, since it handles the case the grain updates subscription as part of `OnNext` call?
BTW, do you have a repro for that whole erros case, or is it just a hunch?